### PR TITLE
Adding support to maintain old and new deploy flow

### PR DESCRIPTION
**Pull request by kushalmalani**
## Description

> Describe the purpose of this pull request.

Adding support to maintain old and new deploy flow

## 🎟 Issue(s)

Related #782 

## 🧪 Functional Testing

> List the functional testing steps to confirm this feature or fix.

## 📸 Screenshots

> Add screenshots to illustrate the validity of these changes.

**After Upgrading (Dag Deploy Feature not enabled) - Old flow just work as expected**
Screen Shot 2022-10-04 at 09 58 55
Screen Shot 2022-10-04 at 10 04 15
Screen Shot 2022-10-04 at 10 05 36
Screen Shot 2022-10-04 at 10 05 52
Screen Shot 2022-10-04 at 10 12 02

**Toggle the flag to enable dag deploy feature and new flow should work**
Screen Shot 2022-10-04 at 10 13 22
Screen Shot 2022-10-04 at 10 13 35
Screen Shot 2022-10-04 at 10 14 26
Screen Shot 2022-10-04 at 10 15 07
Screen Shot 2022-10-04 at 10 15 17

## 📋 Checklist

- [x] Rebased from the main (or release if patching) branch (before testing)
- [x] Ran `make test` before taking out of draft
- [x] Ran `make lint` before taking out of draft
- [x] Added/updated applicable tests
- [ ] Tested against [Astro-API](https://github.com/astronomer/astro/) (if necessary).
- [ ] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: airflow/container.go [hunk 1]
```diff
@@ -39,6 +39,7 @@ type RegistryHandler interface {
 // ImageHandler defines methods require to handle all operations on/for container images
 type ImageHandler interface {
 	Build(config types.ImageBuildConfig) error
+	BuildWithoutDags(config types.ImageBuildConfig) error
```
Comment (neel-astro) [inline airflow/container.go:42]: I think it should be the responsibility of the caller function and not ImageHandler to take care of adding the `dags` folder to dockerignore before building the image. ImageHandler should have logic just related to doing an operation with an image.

In an ideal world, this is how I had imagined the `ImageHandler` & `ContainerHandler` interfaces to be a set of methods that only deals with a particular Container engine for a specific operation related to Image/Container and wouldn't be aware of anything related to airflow logic or in this case `astro deploy`. 

The reasoning behind this was to make sure that we are able to add more container engines like Podman fairly straightforwardly by simply implementing these methods without having to worry about any airflow logic. I think we have already introduced a couple of methods that shouldn't be here and I might have a go at cleaning those up when working on the Podman feature.

**Review by neel-astro — COMMENTED** [review]
have left a minor comment

File: airflow/container.go (hunk 1 above)
Comment (kushalmalani) [reply airflow/container.go:42]: Yeah, that makes sense...let me move that logic to the caller function and update tests

**Review by sunkickr — APPROVED** [review]
LGTM 

only thing that needs to be changed is the `astro deploy update` typo

**Pull request merged**